### DISCUSSION TITLE
Triggering rerender when setting the same value?

### DISCUSSION THREAD

**Discussion by SwatDoge** [vanjs-org/van#372] [Q&A]
## What I have
Currently working on a "time selector", it consists of a few input fields (sec/min/ms) and outputs the combined total of milliseconds on every input.

![Screenshot 2024-10-09 221132](https://github.com/user-attachments/assets/79073579-8143-49a9-8245-de5087253be3)

When this ms total is output, the timeSelector validates this amount, re-converts it into ms/sec/min which re-updates the inputs with correct values.

   ![time selector](https://github.com/user-attachments/assets/9071feb9-fd1d-4009-9f4a-959617867ef3)

## The problem 
When the inputs return more milliseconds than allowed, it gets capped off with the highest allowed value. 
\
The problem is that if the inputs return too many milliseconds *again*, the same max value will be applied again, which causes no re-render to trigger. This means the input fields never get rerendered/cleaned and starts allowing the user to input a bunch of crazy values.

![bug](https://github.com/user-attachments/assets/ce2d04f4-f9f7-430d-be3d-9f591828d19c)

![TimeSelector](https://github.com/user-attachments/assets/c0fe8c58-9fb8-44ba-b4a2-46de8c233bb2)

## Solution?

Is it possible to *always* rerender a state variable when its set, regardless of wether its value changed? Is there a different way to set this up that solves this issue? I'd love to hear

Apologies for the gif/image spam, im not good at explaining.

**Comment by sirenkovladd** [comment]
it seems you just don't register the dependency

try adding something like
```js
const timeVal = time.val;
if (timeSelectorFragmentValues > maxTime)
  time = timeVal;
```

this will allow you to register the dependency regardless of the result of the condition

**Reply by sirenkovladd** [reply]
to deep into, it would be great if you could send an example through [jsfiddle](https://jsfiddle.net/gh/get/library/pure/vanjs-org/vanjs-org.github.io/tree/master/jsfiddle/home/hello)

**Reply by Tao-VanJS** [reply]
Yep, some code would be better for us to understand what's going on.

**Comment by SwatDoge** [comment]
Came back to this project a few weeks later, the problem is rooted way deeper than expected. Apologies for the lack of responses/code examples.
